### ISSUE TITLE
`--resume` ignores the RoPE base in the checkpoint

### ISSUE THREAD

**Issue by felladrin** [felladrin/gguf-trainer#35]
## Problem

`pretrain --resume` and `finetune --resume` do not read the RoPE base from the checkpoint.

Both commands build the model configuration from the command-line flags. Each architecture has a
default RoPE base. If the checkpoint has a different RoPE base, the trainer uses the default value
instead. The trainer shows no warning.

## Effect

The trainer rotates the query and key vectors at the wrong frequency. The model then trains against
position encodings that the model did not learn. The loss gets worse. No message tells the user the
cause.

The defect does not stop at training. `export` writes the wrong RoPE base into the new GGUF file.
llama.cpp reads that wrong value at inference. The published model keeps the defect.

`TinyLlama/TinyLlama_v1.1` is a concrete example. The `config.json` file of TinyLlama_v1.1 has no
`rope_theta` key, so Hugging Face applies the default value 10000. The llama.cpp converter writes
`llama.rope.freq_base = 10000` into the GGUF file. The `llama` architecture in this repository uses
100000 as the flag default. The trainer therefore trains TinyLlama_v1.1 at 100000. That value is 10
times too large. The readme lists TinyLlama_v1.1 as a base model, so a user can meet this defect by
following the documentation.

No published model has the defect. Minueza-3-95M-Base, LittleLamb and SmolLM2-135M-Instruct-heretic
each have a RoPE base that is equal to the default of the architecture of each model. The match is
luck, not design.

## Cause

`src/commands/pretrain.ts:308` builds the configuration with `arch.configFromFlags`.
`configFromFlags` uses the flag default when the user does not give `--rope-base`.

`src/commands/pretrain.ts:321` then compares that configuration against `arch.configFromGGUF`.
`configMatches` compares `vocabSize`, `hiddenSize`, `nLayers`, `nHeads`, `nKVHeads`, `headDim`,
`ffnDim` and `maxSeq`. `configMatches` does not compare `ropeBase`. `configMatches` also does not
compare `ropeBaseLocal`, which gemma3 uses for the sliding-window layers.

`arch.loadWeights` loads the tensors only. `loadWeights` does not change the configuration. The
trainer reads the correct RoPE base from the checkpoint, uses that value for an incomplete
comparison, and then discards that value.

`inspect` prints a "Resume with:" line. That line does not include `--rope-base`. A user who copies
the line gets the defect, because the line looks complete.

## Proposed solution

Make the difference an error, and show the user the flag that corrects the difference. Two changes
are necessary.

**1. Add the RoPE base to the resume gate.**

Add one field to `configMatches` in `src/arch/llama.ts` and in `src/arch/qwen3.ts`:

```ts
{ key: "ropeBase", flag: "rope-base" },
```

Add the same field to `src/arch/gemma3.ts`, and add one more field for the local base:

```ts
{ key: "ropeBaseLocal", flag: "rope-base-local" },
```

`--resume` then stops before the trainer uses the GPU, and the error message names the field. This
behaviour agrees with invariant 2 in `agents.md`: a resume mismatch stops the run and names the
field that is different.

**2. Make `inspect` print the flag that the gate now demands.**

Add `--rope-base` to the "Resume with:" line for all three architectures. Add
`--rope-base-local` for gemma3. A user can then copy one line and resume correctly.

**Do not use this alternative.** An alternative solution is to take the RoPE base from the
checkpoint and to ignore the flag. That alternative hides the difference from the user. That
alternative also prevents context extension, because context extension needs a RoPE base that is
different from the RoPE base of the checkpoint.

## How to test the solution

Add a check to `tests/arch-roundtrip.ts`, or add a new test file:

1. Build a configuration for each architecture.
2. Make a copy of that configuration, and change `ropeBase` in the copy.
3. Give both configurations to `configMatches`.
4. Make sure that `configMatches` returns a message, and make sure that the message contains
   `rope-base`.
5. Do step 2 to step 4 again for `ropeBaseLocal` with gemma3.

Then do a manual test:

1. Download `TinyLlama/TinyLlama_v1.1`.
2. Convert the model with `convert_hf_to_gguf.py`.
3. Run `inspect` on the GGUF file. Make sure that the "Resume with:" line contains
   `--rope-base 10000`.
4. Run `pretrain --resume` with the flags from step 3, but remove `--rope-base`. Make sure that the
   command stops and names `rope-base`.

## One point to check before you close this issue

`diffFields` in `src/arch/common.ts` compares numbers with `!==`. A GGUF file stores the RoPE base
as f32, and the flag value is an f64 in JavaScript. A measurement on two real checkpoints shows that
100000 survives the round trip and stays exactly equal. The common RoPE bases 10000, 100000, 500000
and 1000000 are all exact in f32. A very large value or a value that is not an integer can be
different after the round trip. Test one such value. If the value changes, give `diffFields` a
relative tolerance for numbers that are not integers.

## Background

The `llama` RoPE row-order defect in #32 is a different defect. That defect is fixed. This issue is
about a second RoPE defect that #32 did not touch. An audit of all three architectures found no
other RoPE defect: `gemma3` puts the two RoPE bases on the correct layers with the same phase as
llama.cpp, `qwen3` applies QK-RMSNorm before RoPE as llama.cpp does, the WGSL kernel and the CPU
reference use the same pairing, and no supported checkpoint family uses partial rotary dimensions.
